contracts/earn/maxi/Trisolaris/MaxiVault.sol

```diff
@@ -0,0 +1,253 @@
+import '../../../interfaces/Trisolaris/ITriBar.sol';
+
+contract WOOFiMaxiVault is ERC20Upgradeable, BaseMaxiVault {
+    using SafeERC20 for IERC20;
+    using SafeMathUpgradeable for uint256;
+
+    // Info of each user.
+    struct UserInfo {
+        uint256 amount;     // How many staking tokens the user has provided.
+        uint256 rewardDebt; // Reward debt. See explanation below.
+        //
+        // We do some fancy math here. Basically, any point in time, the amount of reward tokens
+        // entitled to a user but is pending to be distributed is:
+        //
+        //   pending reward = (user.amount * pool.accRewardPerShare) - user.rewardDebt
+        //
+        // Whenever a user deposits or withdraws staking tokens to a pool. Here's what happens:
+        //   1. The pool's `accRewardPerShare` gets updated.
+        //   2. User receives the pending reward sent to his/her address.
+        //   3. User's `amount` gets updated.
+        //   4. User's `rewardDebt` gets updated.
+    }
+
+    /* ----- Mapping ----- */
+
+    mapping (address => UserInfo) public userInfo;
+
+    /* ----- State Variables ----- */
+
+    uint256 public pid;
+    uint256 public accRewardPerShare;
+
+    // Contract Address
+    address public masterChef; // LP Farm
+    address public rewardBar;  // TriBar
+
+    /* ----- Event ----- */
+
+    event Deposit(address indexed user, uint256 amount);
+    event Withdraw(address indexed user, uint256 amount);
+    event ClaimReward(address indexed user, uint256 amount);
+    event Harvest(address indexed user, uint256 amount);
+
+    /* ----- Initializer ----- */
+
+    function initialize(
+        uint256 _pid,
+        address _masterChef,
+        address _rewardBar,
+        address _stakingToken,
+        address _rewardToken,
+        address _wooAccessManager
+    ) external initializer {
+        require(_masterChef != address(0), 'WOOFiMaximizerVault: _masterChef_ZERO_ADDRESS');
+        require(_rewardBar != address(0), 'WOOFiMaximizerVault: _rewardBar_ZERO_ADDRESS');
+        require(_stakingToken != address(0), 'WOOFiMaximizerVault: _stakingToken_ZERO_ADDRESS');
+        require(_rewardToken != address(0), 'WOOFiMaximizerVault: _rewardToken_ZERO_ADDRESS');
+        require(_wooAccessManager != address (0), 'WOOFiMaximizerVault: _wooAccessManager_ZERO_ADDRESS');
+
+        __BaseMaxiVault_init(_wooAccessManager);
+        __ERC20_init(
+            string(abi.encodePacked('WOOFi Earn Maxi ', ERC20Upgradeable(_stakingToken).name())),
+            string(abi.encodePacked('wem', ERC20Upgradeable(_stakingToken).symbol()))
+        );
+
+        pid = _pid;
+        masterChef = _masterChef;
+        rewardBar = _rewardBar;
+        stakingToken = _stakingToken; // LP Token
+        rewardToken = _rewardToken;   // TRI
+
+        _giveAllowances();
+    }
+
+    /* ----- Modifier ----- */
+
+    modifier fairUpdate() {
+        uint256 bal = balance();
```

Comment:
If this case, most of the time is balance is non-zero, just call harvest() then.

Reply:
Agree, update code already.